oak_functions/sdk/oak_functions/src/lib.rs

```diff
@@ -150,12 +154,23 @@ pub fn tf_model_infer(input_vector: &[u8]) -> Result<Inference, OakStatus> {
         )
     };
     let status = OakStatus::from_i32(status_code as i32).ok_or(OakStatus::ErrInternal)?;
-    match status {
-        OakStatus::Ok => {
-            let inference_bytes = from_alloc_buffer(inference_ptr, inference_len);
-            Inference::decode(&*inference_bytes).map_err(|_| OakStatus::ErrInvalidArgs)
-        }
-        status => Err(status),
+
+    if let OakStatus::Ok = status {
+        // We read and interpret the response.
+        let response = from_alloc_buffer(inference_ptr, inference_len);
+        let response: TfModelInferResponse =
+            bincode::deserialize(&response).expect("Fail to deserialize tf response.");
+
+        // We decode the inference bytes.
+        let tf_result = response.result.and_then(|inference_bytes| {
+            Inference::decode(&*inference_bytes)
+                .map_err(|_| TfModelInferError::ErrorDecodingInference)
```

Comment:
Note, here we have a slight change and interpret a decoding error as an TfError, not an OakStatus. Would you disagree?